rlp/src/rlpin.rs

```diff
@@ -260,29 +267,34 @@ impl<'a> Rlp<'a> {
 	}
 
 	/// consumes first found prefix
-	fn consume_list_payload(&self) -> Result<&'a [u8], DecoderError> {
+	fn consume_list_payload(&self) -> Result<(&'a [u8], usize), DecoderError> {
 		let item = BasicDecoder::payload_info(self.bytes)?;
 		if self.bytes.len() < (item.header_len + item.value_len) {
 			return Err(DecoderError::RlpIsTooShort);
 		}
-		Ok(&self.bytes[item.header_len..item.header_len + item.value_len])
+		Ok((&self.bytes[item.header_len..item.header_len + item.value_len], item.header_len))
 	}
 
 	/// consumes fixed number of items
-	fn consume_items(bytes: &'a [u8], items: usize) -> Result<&'a [u8], DecoderError> {
+	fn consume_items(bytes: &'a [u8], items: usize) -> Result<(&'a [u8], usize), DecoderError> {
 		let mut result = bytes;
+		let mut consumed = 0;
 		for _ in 0..items {
 			let i = BasicDecoder::payload_info(result)?;
-			result = Rlp::consume(result, i.header_len + i.value_len)?;
+			let to_consume = i.header_len + i.value_len;
+			result = Rlp::consume(result, to_consume)?;
+			consumed += to_consume;
 		}
-		Ok(result)
+		Ok((result, consumed))
 	}
 
 	/// consumes slice prefix of length `len`
 	fn consume(bytes: &'a [u8], len: usize) -> Result<&'a [u8], DecoderError> {
 		match bytes.len() >= len {
 			true => Ok(&bytes[len..]),
-			false => Err(DecoderError::RlpIsTooShort),
+			false => {
```

Comment:
Is this `rustfmt` doing it's thing? :stuck_out_tongue: I prefer no braces for single-line expressions.